tasks/main.yml

```diff
@@ -1,7 +1,18 @@
 ---
+  - name: Find correct remote user
+    set_fact: remote_user="{{ ansible_ssh_user }}" should_become=true
+    ignore_errors: true
+
+  - name: Find correct remote user
+    set_fact: remote_user="root"  should_become=true
```

Comment:
I mean it probably still work if you use sudo when root, but it's a bit confusing...